substrate/client/rpc-spec-v2/src/chain_head/event.rs

```diff
@@ -39,6 +39,10 @@ pub struct ErrorEvent {
 #[serde(rename_all = "camelCase")]
 pub struct RuntimeVersionEvent {
 	/// The runtime version.
+	#[serde(
+		serialize_with = "custom_serialize::serialize_runtime_version",
+		deserialize_with = "custom_serialize::deserialize_runtime_version"
+	)]
```

Comment:
I do wonder if we could take another approach here, since implementing deserialization is quite complicated in general:
- Create a private `RuntimeVersionWrapper` / `RuntimeVersionInner` which contains the fields in final serailization/deserialization shape
- Use [serde(from = "RuntimeVersionWrapper")](https://serde.rs/container-attrs.html#from) and [serde(into = "RuntimeVersionWrapper")](https://serde.rs/container-attrs.html#into)
- Implement `From` for those structs

Reply:
@lexnv Good idea, would definitely be simpler! I don't know if it is important, but this would come at a slightly higher runtime cost because we would construct a HashMap every time we create a `RuntimeVersionWrapper` from a `RuntimeVersion` before serializing it.